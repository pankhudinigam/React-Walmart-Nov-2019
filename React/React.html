<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">

    <script crossorigin src="https://unpkg.com/react@16/umd/react.development.js"></script>
    <script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-core/5.8.38/browser.js"></script>
    <title>React</title>

</head>

<body>
    <h2>React Application</h2>

    <div id="app1"></div>
    <div id="app2"></div>
    <div id="app3"></div>
    <div id="app4"></div>

    <script type="text/babel">
        var element = React.createElement("h3", null, "This is a React Element");
        ReactDOM.render(element, document.getElementById('app1'));

        var element2 
            = React.createElement("h3", null, 
                            React.createElement("p", null, "This is React element with paragraph"));
        ReactDOM.render(element2, document.getElementById('app2'));

        //JSX
        ReactDOM.render(<h3><p>This is a JSX element</p></h3>, document.getElementById('app3'));
        
        function MyComponent(){
            //JSX
            return (
                <h3><p>This is a React Component</p></h3>
            );
        }

        ReactDOM.render(<MyComponent/>, document.getElementById('app4'));
    
    </script>

</body>

</html>