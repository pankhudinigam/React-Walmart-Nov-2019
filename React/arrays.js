let numbers = [1,2,3,4,5,6];

let results = numbers.filter((item, index) => {
    return item > 3;
})
console.log(results);
results = numbers.map((item, index) => {
    return item * item;
});
console.log(results);

//copy

//ES6 ==> Spread operartor(iterator)
let copy = [ ...numbers ];
numbers[0] = 100;
console.log("copy", copy);
console.log("numbers", numbers);

copy = [ ...numbers, 100, 200, 300 ];
console.log("copy", copy);

let user = {
    id: 100, name: "Anil"
}

let copyUser = {...user, salary: 10000, name: "ANil J"}
console.log(copyUser);

