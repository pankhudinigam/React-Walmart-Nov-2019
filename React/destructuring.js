var array = ["Hello", 100];

// var msg = array[0];
// var id = array[1];


var [msg, id] = array;

console.log(msg);
console.log(id);

var obj = {
    id1: 100, name: "Anil", loc: "Mumbai"
};

var {id1, name, loc} = obj;

console.log(id1);
console.log(name);
console.log(loc);

var test = {
    name
}