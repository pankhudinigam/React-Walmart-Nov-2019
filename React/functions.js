

// function sum(){
//     console.log("in sum no args");
// }

// implicit args => this, arguments
function sum(x, y){
    console.log("in sum: ", arguments);
}

sum(2,3);
sum();
sum(2,3,4,5);
