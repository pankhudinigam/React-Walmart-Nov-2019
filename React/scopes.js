//Design Pattern => IIFE(Immdiatly invoked function expression) => Scope


//Hoisting
console.log("x: ", x);
var x = 10;
console.log("x: ", x);

var y;
console.log("y: ", y);

//console.log("z: ", z);

// function statement
function foo(){
    var msg = "abc";
    console.log("foo..");

    if(x > 1){
        //var test = "Hello scope";
        // (function(){
        //     var test = "Hello scope";
        //     console.log("test", test);
        // })();
        let test = "Hello scope";
        console.log("test", test);
    }
    //console.log("test", test);

}

foo();

//function expresseion
//var bar = function(){console.log();}
//bar();

//iife
(function(){
    console.log("iife");
})();

console.log("App over");