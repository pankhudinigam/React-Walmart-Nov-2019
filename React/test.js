var obj = {
    id: 100,
    print: function(msg){
        console.log("ID: ", this.id)
        console.log("Msg: ", msg)
    }
}

obj.print("abc");

var emp = {
    id: 200
};

//emp.print();

var fn = obj.print.bind(emp);
fn("xyz");

var fn1 = obj.print.bind(emp, "123");

fn1();

obj.print.call(emp, "234");