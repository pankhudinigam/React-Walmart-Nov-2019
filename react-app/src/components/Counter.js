import React, {Component} from 'react';
import SimpleHOC from '../hoc/SimpleHOC';
//bind
// arrow functions(ES6)

class Counter extends Component{

    //ES 7
    state = {
        count: 5
    };
    

    constructor(props){
        super(props);

        this.inc = this.inc.bind(this);
        //ES6
        // this.state = {
        //     count : 5
        // };
    }


    inc = function(evt){
        //console.log("in inc: ", evt);
        //this.state.count++;

        //ASync
        this.setState({
            count: this.state.count + 1
        }, () => {
            console.log(this.state.count);
        });
        
    }
    //ES7
    decr = ()=>{

        this.setState({
            count: this.state.count - 1
        }, () => {
            console.log(this.state.count);
        });
    }

    change = (evt) => {

        this.setState({
            count: parseInt(evt.target.value) 
        });
    }

    update = () => {
        this.setState({
            count: parseInt(this.valueRef.value)
        })
    }
    // setRef = (inpRef) => {
    //     this.valueRef = inpRef;
    // }

    render(){
        
        return (
            <div>
                <h4>Counter =>  {this.props.title}</h4>
                <p>Count: {this.state.count}</p>
                <div>
                    <button id="incBtn" onClick={this.inc}>Increment</button>&nbsp; &nbsp;
                    <button onClick={this.decr}>Decrement</button>
                </div>
                <div>
                    {/* Controlled Input */}
                    Count: <input type="number" 
                            value={this.state.count} onChange={this.change}/>
                </div>
                <div>
                    {/* Uncontrolled Input */}
                    Value: <input type="number" ref={(ref) => {this.valueRef = ref}} /> &nbsp;
                    <button onClick={this.update}>Update</button>
                </div>
                
            </div>
        );
    }
}


//export default SimpleHOC(Counter);
export default Counter;