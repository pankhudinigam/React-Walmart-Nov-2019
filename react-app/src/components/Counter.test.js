import React, {Component} from 'react';
import Counter from './Counter';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, mount, configure} from 'enzyme';
import {create} from 'react-test-renderer';

configure({adapter: new Adapter()})


describe("Counter Test suite", ()=> {

    it("should be created", () => {

        const wrapper =  shallow(<Counter />);
        const instance =  wrapper.instance();
        expect(instance).toBeTruthy();
    });

    it("should have state.count = 5", () => {
        // var x = 10;
        // expect(x).toBe(1);
        const wrapper= mount(<Counter/>);
        const state = wrapper.state();
        const count = state.count;
        expect(count).toBe(5);

    });

    it("should have state.count incremented by 1", () => {
        // var x = 10;
        // expect(x).toBe(1);
        const wrapper= mount(<Counter/>);
        const instance = wrapper.instance();
        instance.inc();
        const state = wrapper.state();
        const count = state.count;
        expect(count).toBe(6);

    });

    it("should have state.count incremented by 1 on button click", () => {
        // var x = 10;
        // expect(x).toBe(1);
        const wrapper= shallow(<Counter />);
        
        wrapper.find("#incBtn").simulate('click');
        wrapper.find("#incBtn").simulate('click');

        const state = wrapper.state();
        const count = state.count;
        expect(count).toBe(7);

    });

    it("snapshot", () => {
       
        const wrapper= create(<Counter title="Counter"/>);
        expect(wrapper.toJSON()).toMatchSnapshot();
    });



})