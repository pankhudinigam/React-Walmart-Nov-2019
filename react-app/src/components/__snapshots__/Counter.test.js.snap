// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Test suite snapshot 1`] = `
<div>
  <h4>
    Counter =&gt;  
    Counter
  </h4>
  <p>
    Count: 
    5
  </p>
  <div>
    <button
      id="incBtn"
      onClick={[Function]}
    >
      Increment
    </button>
       
    <button
      onClick={[Function]}
    >
      Decrement
    </button>
  </div>
  <div>
    Count: 
    <input
      onChange={[Function]}
      type="number"
      value={5}
    />
  </div>
  <div>
    Value: 
    <input
      type="number"
    />
      
    <button
      onClick={[Function]}
    >
      Update
    </button>
  </div>
</div>
`;
